ehrapy/preprocessing/_sampling.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from anndata import AnnData
+
+from anndata import AnnData
+from imblearn.over_sampling import RandomOverSampler
+from imblearn.under_sampling import RandomUnderSampler
+
+
+def sample(
+    adata: AnnData,
+    key: str,
+    random_state: int = 0,
+    method: str = "RandomUnderSampler",
+    sampler_kwargs: dict = None,
+) -> AnnData:
+    """Balancing groups in the dataset.
+
+    Balancing groups in the dataset based on group members in `.obs[key]` using the [imbalanced-learn](https://imbalanced-learn.org/stable/index.html) package.
+    Currently supports `RandomUnderSampler` and `RandomOverSampler`.
+
+    Args:
+        adata: The annotated data matrix of shape `n_obs` × `n_vars`.
+        key: The key in `adata.obs` that contains the group information.
+        random_state: Random seed. Defaults to 0.
+        method: The method to use for balancing. Defaults to "RandomUnderSampler".
+        sampler_kwargs: Keyword arguments for the sample, see the `imbalanced-learn` documentation for options. Defaults to None.
+
+    Returns:
+        A new `AnnData` object, with the balanced groups.
+
+    Example:
+        >>> import ehrapy as ep
+        >>> adata = ep.data.diabetes_130_fairlearn(columns_obs_only=["age"])
+        >>> print("distribution of age groups:\n", adata.obs.age.value_counts())
+        >>> adata_balanced = ep.pp.sample(adata, key="age")
+        >>> print(
```

Comment:
I'd probably remove the prints to keep the examples shorter and more to the point